specs/Qowaiv.Specs/Identifiers/Id_for_Int32_specs.cs

```diff
@@ -49,6 +49,37 @@ public void System_Text_JSON_deserialization(object json, Int32Id svo)
     [TestCase("12345678", 12345678L)]
     public void System_Text_JSON_serialization(Int32Id svo, object json)
         => JsonTester.Write_System_Text_JSON(svo).Should().Be(json);
+
+    [TestCase(-2)]
+    [TestCase(17)]
+    [TestCase("17")]
+    [TestCase(int.MaxValue + 1L)]
+    public void taking_constrains_into_account(object json)
+    {
+        json.Invoking(JsonTester.Read_System_Text_JSON<Id<ForEven>>)
+            .Should().Throw<System.Text.Json.JsonException>()
+            .WithMessage("Not a valid identifier.");
+    }
+
+    private sealed class ForEven : Int32IdBehavior
+    {
+        public override bool TryCreate(object? obj, out object? id)
+        {
+            if(obj is int even && even % 2 == 0)
+            {
+                id = even;
+                return true;
+            }
+            {
```

Comment:
Does this work without else?

Reply:
It was unintended, but yes, it works. `{ .. }` just defines a scope.